rust/operator-binary/src/airflow_controller.rs

```diff
@@ -1415,3 +1448,52 @@ fn add_git_sync_resources(
 
     Ok(())
 }
+
+async fn create_random_secret(
+    secret_name: &str,
+    secret_key: &str,
+    secret_byte_size: usize,
+    airflow: &v1alpha1::AirflowCluster,
+    client: &Client,
+) -> Result<()> {
+    let mut internal_secret = BTreeMap::new();
+    internal_secret.insert(secret_key.to_string(), get_random_base64(secret_byte_size));
+
+    let secret = Secret {
+        immutable: Some(true),
+        metadata: ObjectMetaBuilder::new()
+            .name(secret_name)
+            .namespace_opt(airflow.namespace())
+            .ownerreference_from_resource(airflow, None, Some(true))
+            .context(ObjectMissingMetadataForOwnerRefSnafu)?
+            .build(),
+        string_data: Some(internal_secret),
+        ..Secret::default()
+    };
+
+    if client
+        .get_opt::<Secret>(
+            &secret.name_any(),
+            secret
+                .namespace()
+                .as_deref()
+                .context(ObjectHasNoNamespaceSnafu)?,
+        )
+        .await
+        .context(FailedToRetrieveInternalSecretSnafu)?
+        .is_none()
+    {
+        client
+            .apply_patch(AIRFLOW_CONTROLLER_NAME, &secret, &secret)
+            .await
+            .context(ApplyInternalSecretSnafu)?;
+    }
+
+    Ok(())
+}
+
+fn get_random_base64(byte_size: usize) -> String {
+    let mut buf: Vec<u8> = vec![0; byte_size];
+    openssl::rand::rand_bytes(&mut buf).unwrap();
+    openssl::base64::encode_block(&buf)
+}
```

Comment:
Do we really need openssl for that?

Reply:
I'm with Malte here. Is there no easier way to generate random bytes without using openssl? Especially if we _only_ need openssl for this? (I haven't checked)

Reply:
Alternative (also from Trino): https://github.com/stackabletech/trino-operator/blob/d6943fcca7fc6f08ad08a4fbe20eb555db948455/rust/operator-binary/src/controller.rs#L1502

Reply:
I've replaced it with OsRng here: 7d06d46. It returns a Result, which needs to be handled, but the alternative:
```rust
let mut rng = StdRng::from_os_rng();
rng.fill_bytes(&mut buf);
general_purpose::STANDARD.encode(buf)
```
can *still* panic (according to the docs), the difference begin that the error is not transparent. So I've gone with 

```rust
    OsRng
        .try_fill_bytes(&mut buf)
        .context(SeedRandomGeneratorSnafu)?;
    Ok(general_purpose::STANDARD.encode(buf))
```
to keep things explicit.